src/compiler/crystal/semantic/literal_expander.cr

```diff
@@ -576,6 +564,8 @@ module Crystal
         #     a = temp1
         #     b = temp2
       else
+        raise "BUG: multiple assignment count mismatch" unless node.targets.size == node.values.size
+
```

Comment:
There is now unreachable code just below this `raise`. They shouldn't be there anymow, right?